src/main.rs

```diff
@@ -314,13 +342,21 @@ fn main_send(
         }
 
         let state_clone = state_arc.clone();
+
+        let rl_clone = rl.clone();
         let push_thread = std::thread::spawn(move || {
             let start_time = Instant::now();
             // All the threads iterate through all the files one by one, so all
             // the threads collaborate on sending the first one, then the second
             // one, etc.
+
             'files: for file in state_clone.iter() {
                 'chunks: loop {
+                    if let Some(rl) = rl_clone.lock().unwrap().as_mut() {
+                        let to_wait = rl.time_until_bytes_available(Instant::now(), MAX_CHUNK_LEN);
+                        std::thread::sleep(to_wait);
+                        rl.consume_bytes(Instant::now(), MAX_CHUNK_LEN);
```

Comment:
Now we always count a full chunk, but at the end of the file we send less. When there are few big files this doesn’t matter, but when there are many small files we might limit more than intended. Probably for that case you don’t need Fastsync though so I think it’s fine for now, but also this is the kind of thing that is hard to find later on. Let’s at least add a `FIXME` comment, or alternatively we could pass an `Option<&mut RateLimiter>` to `send_one` and make it consume as much as it sent (or put the amount in `SendResult` and consume it here).

Reply:
This will mean that we must keep the ratelimiter lock while sending, I've implemented it at b99024e3c4e7bd80b841d5fe29dcde76c7ad05c5, let me know what you think